### PR TITLE
feat(nn4k): implement text embeddings

### PR DESCRIPTION
**Pull request by xionghuaidong**

### CODE REVIEW

**Comment by xionghuaidong** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by xionghuaidong** [comment]
recheck

File: python/nn4k/nn4k/invoker/openai.py [hunk 1]
```diff
@@ -35,41 +36,107 @@ def __init__(self, nn_config: dict):
...
-            prompt = [input]
+            prompt = [{"role": "user", "content": input}]
         return prompt
 
+    def _create_completion(self, input, prompt, max_output_length, **kwargs):
+        import openai
+
+        if self._is_legacy_openai_api:
+            completion = openai.ChatCompletion.create(
+                model=self.openai_model_name,
```
Comment (leywar) [inline python/nn4k/nn4k/invoker/openai.py:85]: add spaces on both side of `=`

**Review by leywar — APPROVED** [review]

File: python/nn4k/nn4k/invoker/openai.py (hunk 1 above)
Comment (xionghuaidong) [reply python/nn4k/nn4k/invoker/openai.py:85]: I tried, the Python code formatter **black** will format ``func(a = 1)`` back to ``func(a=1)``.

**Pull request merged**